bftengine/src/bftengine/ReplicaImp.hpp

```diff
@@ -368,7 +369,7 @@ class ReplicaImp : public InternalReplicaApi, public ReplicaForStateTransfer {
   std::function<bool(MessageBase*)> getMessageValidator();
 
   // InternalReplicaApi
-  bool isCollectingState() const override { return stateTransfer->isCollectingState(); }
+  bool isCollectingState() const override { return isCollectingState_; }
```

Comment:
This one is fine, the wrapping of the function saved you the need from replacing all calls.
But I think you've missed something  we've discussed offline - you only set isCollectingState_ on init, but you do not set isCollectingState_ to true when starting ST, so nothing here will really work.
You need to create a wrapper function ReplicaImp::startCollectingState, call stateTransfer->startCollectingState() and set isCollectingState_=true.
Please contact me if you dont fully/clearly understand my explenation.

Reply:
made the changes.